lib/rules/header.js

```diff
@@ -22,17 +22,34 @@ function excludeShebangs(comments) {
     });
 }
 
+// Returns either the first block comment or the first set of line comments that
+// are ONLY separated by a single newline. Note that this does not actually
+// check if they are at the start of the file since that is already checked by
+// hasHeader().
 function getLeadingComments(context, node) {
-    return node.body.length ?
-        context.getComments(node.body[0]).leading :
-        context.getComments(node).leading;
+    var all = excludeShebangs(context.getSourceCode().getAllComments(node.body.length ? node.body[0] : node));
+    if (all[0].type.toLowerCase() === "block") {
+        return [all[0]];
+    } else {
+        for (var i = 1; i < all.length; ++i) {
+            var txt = context.getSourceCode().getText().slice(all[i - 1].range[1], all[i].range[0]);
+            if (!txt.match(/^(\r\n|\r|\n)$/)) {
+                break;
+            }
+        }
+        return all.slice(0, i);
+    }
 }
 
-function genCommentBody(commentType, textArray, eol) {
+function genCommentBody(commentType, textArray, eol, newlines) {
```

Comment:
Is `newlines` a number? Can it be renamed to numNewlines like where it is called?